athena-mysql/src/main/java/com/amazonaws/athena/connectors/mysql/MySqlMetadataHandler.java

```diff
@@ -215,11 +218,47 @@ public GetSplitsResponse doGetSplits(
         return new GetSplitsResponse(getSplitsRequest.getCatalogName(), splits, null);
     }
 
+    private List<TableName> getPaginatedTables(Connection connection, String databaseName, int token, int limit) throws SQLException
+    {
+        String tablesAndViews = "Tables and Views";
+        String sql = "SELECT table_name as \"TABLE_NAME\", table_schema as \"TABLE_SCHEM\" FROM information_schema.tables WHERE table_schema = ? ORDER BY TABLE_NAME LIMIT ?, ?";
+        PreparedStatement preparedStatement = connection.prepareStatement(sql);
+        preparedStatement.setString(1, databaseName);
+        preparedStatement.setInt(2, token);
+        preparedStatement.setInt(3, limit);
+        LOGGER.debug("Prepared Statement for getting tables in schema {} : {}", databaseName, preparedStatement);
+        return JDBCUtil.getTableMetadata(preparedStatement, tablesAndViews);
+    }
+
+    @Override
+    protected ListTablesResponse listPaginatedTables(final Connection connection, final ListTablesRequest listTablesRequest) throws SQLException
+    {
+        String token = listTablesRequest.getNextToken();
+        int pageSize = listTablesRequest.getPageSize();
+
+        // no pagination case
+        if (pageSize == UNLIMITED_PAGE_SIZE_VALUE && token == null) {
+            return new ListTablesResponse(listTablesRequest.getCatalogName(), listTables(connection, listTablesRequest.getSchemaName()), null);
+        }
+
+        // pagination case
+        int t = 0;
+        if (token != null) {
+            t = Integer.valueOf(token);
+        }
+        List<TableName> paginatedTables = getPaginatedTables(connection, listTablesRequest.getSchemaName(), t, pageSize);
+        LOGGER.info("{} tables returned from start token, {}, and end token, {}", paginatedTables.size(), Integer.toString(t), Integer.toString(t + pageSize));
+        ListTablesResponse response = new ListTablesResponse(listTablesRequest.getCatalogName(), paginatedTables, Integer.toString(t + pageSize));
```

Comment:
Can we add unit test or integration test to make sure we the token/pagination logic won't return duplicate? Ideally it should be like
Page size: 20
1st response: [0,20)
2n response: [20,40)